core/domain/opportunity_domain.py

```diff
@@ -22,18 +22,71 @@
 from core import utils
 from core.constants import constants
 
+from typing import Dict, List
+from typing_extensions import TypedDict
+
+
+class PartialExplorationOpportunitySummaryDict(TypedDict):
```

Comment:
Maybe `ExplorationOpportunitySummaryCard` to emphasize that this is the minimum data needed to represent the opportunity on the frontend.

Reply:
@sagangwee  I'm not sure about the `Card` suffix. This class represents a data type not an actual card in the system, right? I'm afraid readers of the code are gonna confuse between domain names and types names if we added the suffix "Card". How about going with a name that this dictionary really represents something like `ExplorationOpportunitySummaryRequiredFields` I know it maybe too long but it expresses exactly what this dictionary is representing